### ISSUE TITLE
Transferwise: dynamic bank account form component

### ISSUE THREAD

**Issue by Betree** [opencollective/opencollective#2858]
With #2836 coming, we'll need a way to input the bank info of the users based on their country. To move faster with this issue, we'll kick of development before design and will see with the design team to make it look better afterwards.

## Best solution for this problem
This component should ideally be self-contained. Something like `BankInfoInput`. To input the country, `InputTypeCountry` should do the job. After that we should fetch the requirements for transferwise and build the form dynamically. `react-hook-form` can be used to make that easier if needed.

A styleguide example would be great.

The component is meant to be used in the [new expense flow](https://www.figma.com/file/ZQBMWhnGGtRWeIZknFW1eP/OC.COM-XX3-%2F-Production-ready?node-id=292%3A3029):

![image](https://user-images.githubusercontent.com/1556356/73527567-90a4ac80-4413-11ea-9eb5-8e7e9e04f74e.png)

## MVP

- [x] Endpoint to fetch necessary fields given a country (proxy to Transferwise).
    - This should take in consideration the host collective we're using since we need its token to request the necessary fields.
- [x] Endpoint to validate the data (proxy to Transferwise).
- [x] Endpoint to create new Payout Method.
- [x] Component that renders necessary fields given the result we got from the backend.
- [ ] Form validation using validation endpoint.
- [x] Create and select payout method when saving the information is successful.

**Comment by Betree** [comment]
@kewitz the most practical for integration would be to implement it in styleguidist (create a `md` file in `styleguide/examples`). The place where it will be displayed is ~350px width max. A controlled component (with `value`, `onChange` and `errors` prop) may be the easiest to integrate.

![image](https://user-images.githubusercontent.com/1556356/74175350-f247f080-4c35-11ea-9cd8-32a2f8dbeca3.png)

**Comment by kewitz** [comment]
@Betree, when should we actually persist the payoutMethod?
Should this form take care of validation and creation of the payoutMethod and return the persisted payoutMethod through the `onChange` function or should I just update the state and call `onChange` for each field?
The catch here is that this form will need to query the backend after a few specific things that are of no concern to the rest of the form. i.e. required fields for certain currency, async validation, etc...

**Comment by Betree** [comment]
> when should we actually persist the payoutMethod?

I'd say we'll persist it when submitting the expense, but we can do it differently if needed.

> Should this form take care of validation and creation of the payoutMethod and return the persisted payoutMethod through the onChange function or should I just update the state and call onChange for each field?
> The catch here is that this form will need to query the backend after a few specific things that are of no concern to the rest of the form. i.e. required fields for certain currency, async validation, etc...

It's not finished yet, but here's the way I've integrated this for other payout methods: https://github.com/opencollective/opencollective-frontend/blob/0cbb56b3626e323109b19f9d63ada131882a00cc/components/expenses/PayoutMethodForm.js

There's also a (not optimized) styleguide example for it (check the first one): https://github.com/opencollective/opencollective-frontend/blob/0cbb56b3626e323109b19f9d63ada131882a00cc/styleguide/examples/expenses/PayoutMethodForm.md

Basically the component doesn't have a local state and calls `onChange` for each field modification. The validate function (`validatePayoutMethod`) is exported from there but is not called from the component - it's the responsibility of to the parent to call it. With that pattern the parent has full control over when the validation is triggered and how (so supporting async validations should not be a problem).

**Comment by kewitz** [comment]
@Betree @znarf related to API calls to TransferWise.
Considering that we need to proxy those calls because the token is private, how should we expose those endpoints?
Through GraphQL or REST?
What namespace or resolver should we use?

**Comment by Betree** [comment]
@kewitz can you provide more details or link to the docs about the endpoint we need to call?

**Comment by kewitz** [comment]
@Betree https://api-docs.transferwise.com/#recipient-accounts-using-account-requirements

**Comment by Betree** [comment]
@kewitz sad that their API forces us to create a quote to achieve this. Anyway we indeed need to abstract this complexity in the API, GraphQL should be the default for that. 

Not sure about the namespace, that's a good question. We need something like a `paymentMethod` namespace with a `bankAccountRequirements` field, but I'm not really sure.

**Comment by kewitz** [comment]
@Betree cool, I agree with that. I'll figure something out and submit for review.
